Sources/Apodini/Semantic Model Builder/Model/ParameterNamespace.swift

```diff
@@ -0,0 +1,107 @@
+//
+// Created by Andi on 06.01.21.
+//
+
+import Foundation
+
+struct ParameterNamespace: OptionSet {
+    let rawValue: UInt8
+
+    static let lightweight = ParameterNamespace(rawValue: 1 << 0)
+    static let content = ParameterNamespace(rawValue: 1 << 1)
+    static let path = ParameterNamespace(rawValue: 1 << 2)
+
+    fileprivate static let all: ParameterNamespace = [.lightweight, .content, .path]
+}
+
+extension ParameterNamespace: CustomStringConvertible {
+    public var description: String {
+        var types: [ParameterType] = []
+
+        if (rawValue & 1) > 0 {
+            types.append(.lightweight)
+        }
+        if (rawValue & 2) > 0 {
+            types.append(.content)
+        }
+        if (rawValue & 4) > 0 {
+            types.append(.path)
+        }
+
+        return types.description
+    }
+}
+
+// MARK: Common ParameterNamespace Definitions
+extension Array where Element == ParameterNamespace {
+    /// With the `.global` level, a parameter name must be
+    /// unique across all `ParameterType`s on the given `Endpoint`.
+    /// This is the default namespace when nothing is specified by the exporter.
+    static let global: [ParameterNamespace] = [.all]
+    /// With the `.individual` level, a parameter name must be
+    /// unique across all parameters with the same `ParameterTyp` on the given `Endpoint`.
+    static let individual: [ParameterNamespace] = [.lightweight, .content, .path]
+}
+
+extension ParameterNamespace {
+    func contains(type: ParameterType) -> Bool {
+        switch type {
+        case .lightweight:
+            return contains(.lightweight)
+        case .path:
+            return contains(.path)
+        case .content:
+            return contains(.content)
+        }
+    }
+}
+
+
+// MARK: Parameter Name Collision
+extension AnyEndpoint {
+    /// Internal method which kicks off the parameter namespace collision checks
+    func parameterNameCollisionCheck(in namespaces: [ParameterNamespace]) {
+        parameters.nameCollisionCheck(on: self, in: namespaces)
+    }
+
+    /// Internal method which kicks off the parameter namespace collision checks
+    func parameterNameCollisionCheck(in namespaces: ParameterNamespace...) {
+        parameters.nameCollisionCheck(on: self, in: namespaces)
+    }
+}
+
+private extension Array where Element == AnyEndpointParameter {
+    func nameCollisionCheck(on endpoint: AnyEndpoint, in namespaces: [ParameterNamespace]) {
+        var namespaces = namespaces
+        if namespaces.isEmpty {
```

Comment:
In case we change the default value to [.global] throwing an error there if an exporter does return an empty collection would be appropriate as this would somehow be a developer error/mistake as the developer would have overwritten the behavior and doesn't provide a strategy.

Reply:
Sounds good to me. I will adjust those parts as described in your other comments as well 👍